### DISCUSSION TITLE
Overriding Internal Components?

### DISCUSSION THREAD

**Discussion by trembl** [vuejs/vitepress#4235] [Q&A]
Hi Everyone,

I am getting an error when I follow the guide on how to [Override Internal Components](https://vitepress.dev/guide/extending-default-theme#overriding-internal-components):

```shell
[vitepress] Internal server error: Failed to resolve import "./VPNavBar.vue" from "node_modules/vitepress/dist/client/theme-default/components/VPNav.vue". Does the file exist?
  Plugin: vite:import-analysis
```

Could anyone actually get _Overriding Internal Components_ to work?

Here is my [minimal test repo](https://github.com/trembl/Vitepress-Test), which produces the error.

- VitePress: 1.3.4
- Node: v20.17.0

Thank you!

**Comment by Ragnar-Oock** [comment]
The replacement path provided in the doc is incorrect, you need to add `./theme` at the start of the replacement path :

```diff
{
	find: /^.*\/VPNavBar\.vue$/,
	replacement: fileURLToPath(
-		new URL('./components/CustomNavBar.vue', import.meta.url)
+		new URL('./theme/components/CustomNavBar.vue', import.meta.url)
	)
}
